invenio_rdm_records/resources/serializers/marcxml/schema.py

```diff
@@ -0,0 +1,287 @@
+
+from ..ui.schema import current_default_locale
+from ..utils import get_vocabulary_props
+
+
+class MARCXMLSchema(Schema):
+    """Schema for records in MARC."""
+
+    contributors = fields.Method("get_contributors")
+    titles = fields.Method("get_titles")
+    creators = fields.Method("get_creators")
+    identifiers = fields.Method("get_identifiers")
+    relations = fields.Method("get_relations")
+    rights = fields.Method("get_rights")
+    dates = fields.Method("get_dates")
+    subjects = fields.Method("get_subjects")
+    descriptions = fields.Method("get_descriptions")
+    publishers = fields.Method("get_publishers")
+    types = fields.Method("get_types")
+    sources = fields.Method("get_sources")
+    coverage = fields.Method("get_locations")
+    formats = fields.Method("get_formats")
+    parent_id = fields.Method("get_id")  # No GOOD
+    community_id = fields.Method("get_id_community")
+    last_updated = fields.Method("last_updated")
+    sizes = fields.Method("get_sizes")
+    version = fields.Method("get_version")
+    funding = fields.Method("get_funding")
+
+    def get_version(self, obj):
+        """Get the version of the record."""
+        return obj["metadata"].get("version", missing)
+
+    def get_sizes(self, obj):
+        """Get the size of the record."""
+        return obj["metadata"].get("sizes", missing)
+
+    def get_funding(self, obj):
+        """Get funding information."""
+        if "funding" not in obj["metadata"]:
+            return missing
+
+        funders = []
+
+        for funder in obj["metadata"].get("funding", []):
+            funder_string = ""
+
+            scheme = (
+                funder.get("award", {})
+                .get("identifiers", [{}])[0]
+                .get("scheme", "null")
+            )
+            identifier = (
+                funder.get("award", {})
+                .get("identifiers", [{}])[0]
+                .get("identifier", "null")
+            )
+
+            title = funder.get("award", {}).get("title", {})
+            title = list(title.values())[0] if title != {} else "null"
+
+            number = funder.get("award", {}).get("number", "null")
+            id = funder["funder"]["id"]
+            name = funder["funder"].get("name", "null")
+
+            funder_string += f"award_identifiers_scheme={scheme}; "
+            funder_string += f"award_identifiers_identifier={identifier}; "
+            funder_string += f"award_title={title}; "
+            funder_string += f"award_number={number}; "
+            funder_string += f"funder_id={id}; "
+            funder_string += f"funder_name={name}; "
+            funders.append(funder_string)
+
+        return funders
+
+    def last_updated(self, obj):
+        """Get date last updated."""
+        return [obj["updated"]]
+
+    def get_locations(self, obj):
+        """Get locations."""
+        locations = []
+        if not ("metadata" in obj and "locations" in obj["metadata"]):
```

Comment:
why do we check for `metadata in obj` here, but in all the other fields we do `obj["metadata"]`? My question: this gives the impression that metadata could not be there, however if that was the case it would throw a `KeyError` in any of the other functions. Therefore, is the first condition of the `if` needed?

Reply:
you're correct this is redundant, will fix